CBinding/Compiler/CMakeToolchain.cs

```diff
@@ -0,0 +1,228 @@
+		{
+			var results = new BuildResult ();
+			result.Position = 0;
+			var sr = new StreamReader (result);
+			var sb = new StringBuilder ();
+			string line;
+			string fileName = "";
+			int lineNumber = 0;
+			bool isWarning = false;
+
+			while ((line = sr.ReadLine ()) != null) {
+				//e.g.	CMake Warning in/at CMakeLists.txt:10 (COMMAND):
+				//or:	CMake Warning:
+				if (line.StartsWith ("CMake Warning", StringComparison.OrdinalIgnoreCase)) {
+					//reset everything and add last error or warning.
+					if (sb.Length > 0) {
+						if (isWarning)
+							results.AddWarning (fileName, lineNumber, 0, "", sb.ToString ());
+						else
+							results.AddError (fileName, lineNumber, 0, "", sb.ToString ());
+					}
+
+					sb.Clear ();
+					fileName = "";
+					lineNumber = 0;
+					isWarning = true;
+
+					// in/at CMakeLists.txt:10 (COMMAND):
+					if (line.Contains (" in ")) {
+						Tuple<int, string> t = GetFileAndLine (line, " in ");
+						lineNumber = t.Item1;
+						fileName = t.Item2;
+					} else if (line.Contains (" at ")) {
+						Tuple<int, string> t = GetFileAndLine (line, " at ");
+						lineNumber = t.Item1;
+						fileName = t.Item2;
+					} else {
+						string [] warning = line.Split (':');
+						if (!string.IsNullOrEmpty (warning.ElementAtOrDefault (1))) {
+							sb.Append (warning [1]);
+						}
+					}
+				} else if (line.StartsWith ("CMake Error", StringComparison.OrdinalIgnoreCase)) {
+					//reset everything and add last error or warning.
+					if (sb.Length > 0) {
+						if (isWarning)
+							results.AddWarning (fileName, lineNumber, 0, "", sb.ToString ());
+						else
+							results.AddError (fileName, lineNumber, 0, "", sb.ToString ());
+					}
+
+					sb.Clear ();
+					fileName = "";
+					lineNumber = 0;
+					isWarning = false;
+
+					// in/at CMakeLists.txt:10 (COMMAND):
+					if (line.Contains (" in ")) {
+						Tuple<int, string> t = GetFileAndLine (line, " in ");
+						lineNumber = t.Item1;
+						fileName = t.Item2;
+					} else if (line.Contains (" at ")) {
+						Tuple<int, string> t = GetFileAndLine (line, " at ");
+						lineNumber = t.Item1;
+						fileName = t.Item2;
+					} else {
+						string [] error = line.Split (':');
+						if (!string.IsNullOrEmpty (error.ElementAtOrDefault (1))) {
+							sb.Append (error [1]);
+						}
+					}
+				} else {
+					sb.Append (line);
+				}
+			}
+
+			return results;
+		}
+
+		public static CMakeToolchain GetToolchain ()
+		{
+			string toolchainName;
+			OperatingSystem os = Environment.OSVersion;
+			if (os.Platform == PlatformID.Win32Windows || os.Platform == PlatformID.Win32S || os.Platform == PlatformID.WinCE || os.Platform == PlatformID.Win32NT) {
+				toolchainName = PropertyService.Get<string> ("CBinding.DefaultToolchain", new MinGW32Toolchain ().ToolchainName);
+				return AddinManager.GetExtensionObjects<CMakeToolchain> ("/CBinding/Toolchains").FirstOrDefault<CMakeToolchain> (toolchain => toolchain.ToolchainName == toolchainName);
```

Comment:
This could return null in some cases.

I would probably factor out a GetDefaultToolchain() method that returns the correct toolchain for the current platform. Then do something like:

```csharp
toolchainName = PropertyService.Get<string> ("CBinding.DefaultToolchain", null);
return AddinManager.GetExtensionObjects<CMakeToolchain> ("/CBinding/Toolchains")
    .FirstOrDefault<CMakeToolchain> (toolchain => toolchain.ToolchainName == toolchainName)
    ?? GetDefaultToolchain();
```